### PR TITLE
feat: simplify discovery preparation web flow

### PR DESCRIPTION
**Pull request by ebarti**
## Summary

- Present `Discover` as the single user-facing preparation action in pipeline controls, dashboard funnel, jobs table stages, and job detail diagnostics.
- Add context-owned rescore and re-tailor mutation hooks/buttons for per-job, selected-job, and outdated-policy maintenance flows using the API/projection contracts from the stacked base.
- Treat suppressed artifacts as historical audit material in artifact status copy, filtering, and detail UI, rather than active apply-ready material.
- Update web/MSW/unit/E2E coverage around the new maintenance controls, product-stage display, artifact suppression state, and current data-grid/debug views.

## Why

The discovery preparation flow now needs one operator-facing preparation action while keeping score and tailoring work visible as diagnostics and maintenance. This keeps the UI aligned with the single preparation stage model without changing worker or API contracts in this PR.

## Validation

- `pnpm web:check` — passed
- `pnpm --filter @jobhunter/web test-d` — passed, 10 files / 10 tests, no type errors
- `pnpm --filter @jobhunter/web test -- src/views/dashboard/Funnel.test.tsx` — passed, 107 files / 545 tests
- `pnpm --filter @jobhunter/web e2e` — passed, 8 passed / 1 existing skipped materials spec
- `git diff --check feat/discovery-preparation-api-projections...HEAD` — passed

## Documentation

No documentation update is included because this PR only wires the web product flow to contracts already introduced by the stacked base and does not change public CLI/runtime behavior.

### CODE REVIEW

File: apps/web/src/views/dashboard/Funnel.tsx [hunk 1]
```diff
@@ -9,21 +9,102 @@ export interface FunnelProps {
...
+
+function buildDiscoverStage(summary: DashboardSummary): ProductFunnelStage | null {
+  const preparationStages = summary.funnel.filter((stage) =>
+    PREPARATION_STAGES.has(stage.stage),
+  );
+  if (!preparationStages.length && !summary.preparation) {
+    return null;
+  }
+
+  const failed =
```
Comment (ebarti) [inline apps/web/src/views/dashboard/Funnel.tsx]: PR Reviewer - Codex

Severity: High
Gate: Fail
Evidence: `apps/web/src/views/dashboard/Funnel.tsx:52-60` uses `summary.preparation.workItems` whenever `preparation` exists, and `apps/api/src/read-model.ts:291` always includes `preparation`. That makes the `sumStages(preparationStages, ...)` fallback unreachable in normal API responses, so score/tailor/cover funnel failures and running/pending counts disappear from the single Discover row. The changed fixture already has `totals.failures = 1` and `score.failed = 1` while `preparation.workItems.failed = 0`, so the new funnel renders no failed Discover diagnostic for that failed score stage.
Reproduction: Reasoned from code; `pnpm web:check` passed, `pnpm --filter @jobhunter/web test -- ...` ran the web Vitest suite and passed, and `pnpm --filter @jobhunter/web test-d` passed.

The PR goal is to move scoring/tailoring into diagnostic substatus, not hide their failed/running/pending state. Because the API always supplies `preparation`, the dashboard now replaces the internal preparation-stage funnel state with only `preparation_work_items` counts, which are a separate maintenance queue. Operators can lose the visible Discover-row failure signal for score/tailor/cover jobs.

Required fix: Aggregate the internal preparation stage funnel counts into the Discover row even when `summary.preparation` exists, and surface `preparation.workItems` as an additional maintenance diagnostic if needed. Add a regression test where `summary.funnel` has a score/tailor failure but `preparation.workItems.failed` is zero and assert the Discover row still reports the failure.

**Review by ebarti — COMMENTED** [review]
I found one user-visible issue in the dashboard product funnel aggregation.

File: apps/web/src/views/dashboard/Funnel.tsx (hunk 1 above)
Comment (ebarti) [reply apps/web/src/views/dashboard/Funnel.tsx]: Fixed in `7aa8128`. `buildDiscoverStage` now folds the hidden preparation-stage funnel counts into the Discover row even when `summary.preparation.workItems` exists, adding work-item queued/running/failed counts instead of replacing score/tailor/cover state. Added a focused regression in `Funnel.test.tsx` that reproduces a hidden score/tailor/cover failure/running/pending case with zero prep work-item failures.

Verification:
- `git diff --check feat/discovery-preparation-api-projections...HEAD` — passed
- `pnpm --filter @jobhunter/web test -- src/views/dashboard/Funnel.test.tsx` — passed, 107 files / 545 tests
- `pnpm web:check` — passed
- `pnpm --filter @jobhunter/web test-d` — passed, 10 files / 10 tests, no type errors

**Comment by ebarti** [comment]
PR Fixer - Codex

Fixed the High Discover funnel aggregation finding from thread `PRRT_kwDOSQ-EBM6E9ivv` / comment `3307679039`.

- `apps/web/src/views/dashboard/Funnel.tsx` now aggregates hidden Discover/score/tailor/cover funnel counts into the single Discover row even when `summary.preparation.workItems` exists. Work-item queued/running/failed counts are added instead of replacing stage counts.
- `apps/web/src/views/dashboard/Funnel.test.tsx` adds a regression where prep work items exist with zero failures while hidden score/tailor/cover stages still expose failed/running/pending Discover state.

Verification:
- `git diff --check feat/discovery-preparation-api-projections...HEAD` - passed
- `pnpm --filter @jobhunter/web test -- src/views/dashboard/Funnel.test.tsx` - passed, 107 files / 545 tests
- `pnpm web:check` - passed
- `pnpm --filter @jobhunter/web test-d` - passed, 10 files / 10 tests, no type errors

Dismissed findings: None.
Remaining Blocker/High findings: None.

File: apps/web/src/views/jobs/columns.tsx [hunk 2]
```diff
@@ -236,7 +240,15 @@ export function jobColumns(
       label: "Stage",
       sortable: true,
       getFilterValue: (row) => row.currentStage,
-      render: (row) => <StageBadge stage={row.currentStage} />,
+      render: (row) => {
```
Comment (ebarti) [inline apps/web/src/views/jobs/columns.tsx:243]: PR Reviewer - Codex

Severity: High
Gate: Fail
Evidence: `apps/web/src/views/jobs/columns.tsx:243-249` maps internal preparation stages to `Discover` in the table cell, but `apps/web/src/views/jobs/JobFilterBar.tsx:40-49` still renders every `STAGES` option directly, so the Jobs toolbar still exposes `enrich`, `score`, `tailor`, and `cover` as primary stage filters. `apps/web/src/views/jobs/JobFilterBar.test.tsx:41-59` still asserts selecting `tailor` and rendering one option per internal stage.
Reproduction: Reproduced with command: `pnpm --filter @jobhunter/web test -- src/views/jobs/JobFilterBar.test.tsx src/views/jobs/JobBulkActions.test.tsx src/views/dashboard/Funnel.test.tsx` passed while preserving the internal stage filter behavior.

The PR goal says `Discover` is the single user-facing preparation action and score/tailor/cover move to diagnostic substatus/maintenance controls. Masking the table cell alone leaves the list's primary stage filter in the old product model, so users can still navigate the main Jobs view by raw internal preparation stages.

Required fix: Stop rendering raw internal preparation stages as user-facing Jobs filter options. Replace them with product-stage options (`all`, `discover`, `apply`) or clearly present any internal-stage narrowing as Discover diagnostic substatus, and update `JobFilterBar.test.tsx` so it no longer requires raw `score`/`tailor`/`cover` options.

File: apps/web/src/contexts/scoring/components/RescoreCurrentPolicyButton.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      disabled={blocked}
+      title={jobKeys.length ? undefined : `${count} jobs are not on the current scoring policy.`}
+      type="button"
+      onClick={() => {
+        if (blocked) return;
+        const scope = jobKeys.length
+          ? `${count} selected job${count === 1 ? "" : "s"}`
+          : `up to ${limit} jobs not on the current scoring policy`;
+        if (!window.confirm(`Rescore ${scope}?`)) return;
+        mutation.mutate({ jobKeys, limit }, onSuccess ? { onSuccess } : undefined);
```
Comment (ebarti) [inline apps/web/src/contexts/scoring/components/RescoreCurrentPolicyButton.tsx]: PR Reviewer - Codex

Severity: Medium
Gate: Pass
Evidence: `apps/web/src/contexts/scoring/components/RescoreCurrentPolicyButton.tsx:53,75-79` defaults `limit` to 100 and sends that limit even when `jobKeys` is the explicit selected set; `apps/web/src/contexts/materials/components/RetailorCurrentPolicyButton.tsx:58,80-90` does the same for re-tailor. Jobs search allows `pageSize` up to 200 (`apps/web/src/routes/-jobs.search.ts:17`), and the current-policy activities apply the limit while selecting explicit job URLs (`workers/automation/src/jobhunter/scoring/activities.py:108-114`, `workers/automation/src/jobhunter/materials/activities.py:124-131`). A selected page with 101-200 rows can therefore show `rescore selected` / `re-tailor selected` but enqueue only the first 100 eligible selected URLs.
Reproduction: Reasoned from code; focused Vitest command passed but does not cover a selected set larger than the default maintenance limit.

The selected buttons promise to operate on the selected jobs, while the default maintenance cap is meant for the unscoped outdated-policy batch. Reusing that cap for explicit selections creates a silent partial bulk action.

Required fix: When `jobKeys.length > 0`, send a limit that covers the selected set (for example `Math.min(jobKeys.length, 1000)`) or otherwise bypass the unscoped batch cap for explicit selections. Add focused tests for more than 100 selected job keys for both rescore and re-tailor.

**Review by ebarti — COMMENTED** [review]
I found one remaining phase-goal issue in the Jobs product flow and one selected-bulk edge case.

File: apps/web/src/views/jobs/columns.tsx (hunk 2 above)
Comment (ebarti) [reply apps/web/src/views/jobs/columns.tsx:243]: Fixed in `c52fa0b`. `JobFilterBar` now renders only the product-stage filter options (`all`, `discover`, `apply`) instead of mapping over raw internal stages. `JobFilterBar.test.tsx` now asserts the exact product option set and rejects `enrich`/`score`/`tailor`/`cover` as user-facing stage options.

Verification:
- `pnpm --filter @jobhunter/web test -- src/views/jobs/JobFilterBar.test.tsx src/views/jobs/JobBulkActions.test.tsx src/contexts/scoring/components/RescoreCurrentPolicyButton.test.tsx src/contexts/materials/components/RetailorCurrentPolicyButton.test.tsx src/contexts/scoring/hooks/useRescoreCurrentPolicyMutation.test.ts src/contexts/materials/hooks/useRetailorCurrentPolicyMutation.test.ts` - passed, 107 files / 547 tests
- `pnpm web:check` - passed
- `pnpm --filter @jobhunter/web test-d` - passed, 10 files / 10 tests, no type errors
- `git diff --check feat/discovery-preparation-api-projections...HEAD` - passed

File: apps/web/src/contexts/scoring/components/RescoreCurrentPolicyButton.tsx (hunk 3 above)
Comment (ebarti) [reply apps/web/src/contexts/scoring/components/RescoreCurrentPolicyButton.tsx]: Fixed in `c52fa0b`. Selected rescore and re-tailor actions now send a request limit at least as large as the explicit `jobKeys` selection, and the mutation request builders enforce the same guard before calling the API. `JobBulkActions.test.tsx` covers 101 selected rows for both selected rescore and selected re-tailor payloads.

Verification:
- `pnpm --filter @jobhunter/web test -- src/views/jobs/JobFilterBar.test.tsx src/views/jobs/JobBulkActions.test.tsx src/contexts/scoring/components/RescoreCurrentPolicyButton.test.tsx src/contexts/materials/components/RetailorCurrentPolicyButton.test.tsx src/contexts/scoring/hooks/useRescoreCurrentPolicyMutation.test.ts src/contexts/materials/hooks/useRetailorCurrentPolicyMutation.test.ts` - passed, 107 files / 547 tests
- `pnpm web:check` - passed
- `pnpm --filter @jobhunter/web test-d` - passed, 10 files / 10 tests, no type errors
- `git diff --check feat/discovery-preparation-api-projections...HEAD` - passed

**Comment by ebarti** [comment]
PR Fixer - Codex

Fixed findings:
- High thread `PRRT_kwDOSQ-EBM6E9q_y` / comment `3307725310`: `JobFilterBar` now renders only `all`, `discover`, and `apply` product-stage options; `JobFilterBar.test.tsx` asserts the exact option set and rejects raw internal preparation stages.
- Medium thread `PRRT_kwDOSQ-EBM6E9q_z` / comment `3307725311`: selected rescore and re-tailor actions now send a limit at least as large as the explicit selected `jobKeys`; the mutation request builders enforce the same guard; `JobBulkActions.test.tsx` covers 101 selected jobs for both actions.

Dismissed findings: None.

Verification:
- `pnpm --filter @jobhunter/web test -- src/views/jobs/JobFilterBar.test.tsx src/views/jobs/JobBulkActions.test.tsx src/contexts/scoring/components/RescoreCurrentPolicyButton.test.tsx src/contexts/materials/components/RetailorCurrentPolicyButton.test.tsx src/contexts/scoring/hooks/useRescoreCurrentPolicyMutation.test.ts src/contexts/materials/hooks/useRetailorCurrentPolicyMutation.test.ts` - passed, 107 files / 547 tests
- `pnpm web:check` - passed
- `pnpm --filter @jobhunter/web test-d` - passed, 10 files / 10 tests, no type errors
- `git diff --check feat/discovery-preparation-api-projections...HEAD` - passed

Remaining Blocker/High findings: None.

File: apps/web/src/views/jobs/JobFilterBar.tsx [hunk 4]
```diff
@@ -1,8 +1,13 @@
-import { STAGES, STAGE_STATES } from "@jobhunter/contracts";
+import { STAGE_STATES } from "@jobhunter/contracts";
 import { useNavigate } from "@tanstack/react-router";
 
 import type { JobsSearch } from "../../routes/-jobs.search.js";
 
+const STAGE_OPTIONS = [
+  { value: "all", label: "all stages" },
+  { value: "discover", label: "discover" },
```
Comment (ebarti) [inline apps/web/src/views/jobs/JobFilterBar.tsx:8]: PR Reviewer - Codex

Severity: High
Gate: Fail
Evidence: `apps/web/src/views/jobs/JobFilterBar.tsx:6-10` exposes `discover` as the single user-facing preparation filter, but `apps/web/src/views/jobs/JobsView.tsx:20-31` and `apps/web/src/routes/jobs.tsx:7-17` pass `stage: "discover"` directly into the jobs API input. The API read model still treats `stage` as an exact internal stage (`apps/api/src/read-model.ts:1292-1294` and `:1478-1480`), so choosing Discover excludes jobs whose current internal substatus is `enrich`, `score`, `tailor`, or `cover` even though `apps/web/src/views/jobs/columns.tsx:235-246` displays those rows as Discover.
Reproduction: Reasoned from code.

The product filter now claims to represent the whole preparation bucket, but the query still narrows to only literal `discover`. Users filtering to Discover will lose the score/tailor/cover preparation jobs they are supposed to see as diagnostic substatus, and bulk actions against "all matching" on that filter inherit the same under-selection.

Required fix: Translate the product-stage filter before querying/mutating so `discover` includes all preparation stages (`discover`, `enrich`, `score`, `tailor`, `cover`), or add a dedicated product-stage query/filter contract. Add a JobsView or route-level regression proving the Discover filter request includes score/tailor/cover preparation jobs rather than only literal `discover`.

**Review by ebarti — COMMENTED** [review]
I found one remaining product-flow issue in the Jobs Discover filter semantics.

File: apps/web/src/views/jobs/JobFilterBar.tsx (hunk 4 above)
Comment (ebarti) [reply apps/web/src/views/jobs/JobFilterBar.tsx:8]: Fixed in `a427296`. The Jobs product-stage mapper now expands the user-facing `discover` filter to the preparation-stage set (`discover`, `enrich`, `score`, `tailor`, `cover`) before list loading, and `useJobsListQuery` resolves that multi-stage input by querying exact internal stages and combining the paginated response. `apply`, `all`, and manually supplied internal stages remain exact/single-filter behavior.

Bulk all-matching filters now use the same mapper, so Discover fans out one all-matching mutation filter per preparation stage instead of sending only `stage: "discover"`.

Regression coverage added in `JobsView.test.tsx` for:
- score/tailor/enrich/cover rows appearing under product Discover filtering.
- Apply remaining an exact `apply` query.
- all-matching Discover bulk filters fanning out across preparation stages.

Verification:
- `pnpm --filter @jobhunter/web test -- src/views/jobs/JobsView.test.tsx src/views/jobs/JobFilterBar.test.tsx` - passed, 107 files / 550 tests
- `pnpm web:check` - passed
- `pnpm --filter @jobhunter/web test-d` - passed, 10 files / 10 tests, no type errors
- `git diff --check feat/discovery-preparation-api-projections...HEAD` - passed

**Comment by ebarti** [comment]
PR Fixer - Codex

Fixed findings:
- High thread `PRRT_kwDOSQ-EBM6E9w09` / comment `3307758382`: product Discover now maps to the preparation stages before Jobs list loading and all-matching bulk filtering. `useJobsListQuery` resolves multi-stage inputs by querying exact internal stages and combining the paginated result. `apply`, `all`, and manually supplied internal stages remain exact/single-filter behavior.

Dismissed findings: None.

Verification:
- `pnpm --filter @jobhunter/web test -- src/views/jobs/JobsView.test.tsx src/views/jobs/JobFilterBar.test.tsx` - passed, 107 files / 550 tests
- `pnpm web:check` - passed
- `pnpm --filter @jobhunter/web test-d` - passed, 10 files / 10 tests, no type errors
- `git diff --check feat/discovery-preparation-api-projections...HEAD` - passed

Remaining Blocker/High findings: None.
